src/features/cookies/useCookies.test.ts

```diff
@@ -0,0 +1,128 @@
+import mockdate from 'mockdate'
+
+import { ALL_OPTIONAL_COOKIES, COOKIES_BY_CATEGORY } from 'features/cookies/CookiesPolicy'
+import { COOKIES_CONSENT_KEY, useCookies } from 'features/cookies/useCookies'
+import { storage } from 'libs/storage'
+import { act, renderHook, waitFor } from 'tests/utils'
+
+const deviceId = 'testUuidV4'
+const Today = new Date(2022, 9, 29)
+mockdate.set(Today)
+
+describe('useCookies', () => {
+  beforeEach(() => storage.clear(COOKIES_CONSENT_KEY))
+
+  describe('state', () => {
+    it('should have all cookies consent refused by default', () => {
+      const { result } = renderHook(useCookies)
+      const { cookiesChoice } = result.current
+
+      expect(cookiesChoice.consent).toEqual({
+        mandatory: COOKIES_BY_CATEGORY.essential,
+        accepted: [],
+        refused: ALL_OPTIONAL_COOKIES,
+      })
+    })
+
+    it('should set device ID', () => {
+      const { result } = renderHook(useCookies)
+      const { cookiesChoice } = result.current
+
+      expect(cookiesChoice.deviceId).toEqual(deviceId)
+    })
+
+    it('should not set choiceDateTime by default', () => {
+      const { result } = renderHook(useCookies)
+      const { cookiesChoice } = result.current
+
+      expect(cookiesChoice.choiceDatetime).toEqual(undefined)
+    })
+
+    it('should write state', () => {
+      const { result, rerender } = renderHook(useCookies)
+      const { setCookiesChoice } = result.current
+
+      act(() => {
+        setCookiesChoice({
+          deviceId,
+          choiceDatetime: Today,
+          consent: {
+            mandatory: COOKIES_BY_CATEGORY.essential,
+            accepted: ALL_OPTIONAL_COOKIES,
+            refused: [],
+          },
+        })
```

Comment:
ça serait cool que le hook expose uniquement

```suggestion
        setCookiesConsent({
            mandatory: COOKIES_BY_CATEGORY.essential,
            accepted: ALL_OPTIONAL_COOKIES,
            refused: [],
        })
```

Reply:
Mmmmh je vois pas pourquoi faire ça ? On veut séparer les informations ?